extensions/podman/src/podman-install.ts

```diff
@@ -294,3 +310,97 @@ class WinInstaller extends BaseInstaller {
     });
   }
 }
+
+abstract class BaseCheck implements extensionApi.InstallCheck {
+  abstract title: string;
+  abstract execute(): Promise<extensionApi.CheckResult>;
+
+  protected createFailureResult(description?: string): extensionApi.CheckResult {
+    return { successful: false, description };
+  }
+
+  protected createSuccessfulResult(): extensionApi.CheckResult {
+    return { successful: true };
+  }
+}
+
+class WinBitCheck extends BaseCheck {
+  title = 'Windows x64';
+  private ARCH = 'x64';
+  async execute(): Promise<extensionApi.CheckResult> {
+    const currentArch = process.arch;
+    if (this.ARCH === currentArch) {
+      return this.createSuccessfulResult();
+    } else {
+      return this.createFailureResult('WSL2 works only on x64 OS.');
```

Comment:
I don't know if it would be possible to add links/pointers to the user explaining it with more details (could be another iteration)

like showing https://docs.microsoft.com/en-us/windows/wsl/install or stuff like that

Here what is unclear to me is that it seems WL 2 is working with ARM64 based on https://docs.microsoft.com/en-us/windows/wsl/install-manual#step-2---check-requirements-for-running-wsl-2

so here probably it doesn't work for Podman, but not for WSL2 (need to change the failure message ?)